libs/cardano-ledger-api/src/Cardano/Ledger/Api/Tx.hs

```diff
@@ -71,4 +78,76 @@ import Cardano.Ledger.Api.Tx.AuxData
 import Cardano.Ledger.Api.Tx.Body
 import Cardano.Ledger.Api.Tx.Cert
 import Cardano.Ledger.Api.Tx.Wits
-import Cardano.Ledger.Core (EraTx (..), setMinFeeTx)
+import Cardano.Ledger.BaseTypes (ProtVer (..))
+import Cardano.Ledger.Binary (byronProtVer, decodeFull', serialize')
+import Cardano.Ledger.Coin (Coin)
+import Cardano.Ledger.Core (EraCrypto, EraTx (..), PParams, ppProtocolVersionL, setMinFeeTx)
+import Cardano.Ledger.Crypto
+import Cardano.Ledger.Keys (BootstrapWitness (..), ChainCode (..))
+import Data.Bits (shiftR)
+import qualified Data.ByteString as BS
+import Data.Proxy
+import qualified Data.Set as Set
+import Lens.Micro
+
+-- | Estimate a minimum transaction fee for a transaction that does not yet have all of
+-- the `VKey` witnesses. This calculation is not very accurate in estimating Byron
+-- witnesses, but it should work for the most part.
+--
+-- @since 1.8.0
+estimateMinFeeTx ::
+  forall era.
+  EraTx era =>
+  -- | The current protocol parameters.
+  PParams era ->
+  -- | The transaction.
+  Tx era ->
+  -- | The number of key witnesses still to be added to the transaction.
+  Int ->
```

Comment:
How come this is changed to `Int`? Wouldn't `Word` (as it is in the deprecated function) be a better fit for this type, since it's unsigned?

Reply:
Oh interesting. Thanks for the thorough explanation